Mage/src/main/java/mage/abilities/effects/common/ExileCardsFromTopOfLibraryControllerEffect.java

```diff
@@ -0,0 +1,103 @@
+package mage.abilities.effects.common;
+
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Card;
+import mage.constants.Outcome;
+import mage.game.Game;
+import mage.players.Player;
+import mage.util.CardUtil;
+
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ * @author Cguy7777
+ */
+public class ExileCardsFromTopOfLibraryControllerEffect extends OneShotEffect {
+
+    private final int amount;
+    private final boolean toUniqueExileZone;
+    private final boolean faceDown;
+    private final boolean withFaceDownReminderText;
+
+    public ExileCardsFromTopOfLibraryControllerEffect(int amount) {
+        this(amount, false);
+    }
+
+    public ExileCardsFromTopOfLibraryControllerEffect(int amount, boolean toUniqueExileZone) {
+        this(amount, toUniqueExileZone, false);
+    }
+
+    public ExileCardsFromTopOfLibraryControllerEffect(int amount, boolean toUniqueExileZone, boolean faceDown) {
+        this(amount, toUniqueExileZone, faceDown, false);
+    }
+
+    /**
+     * @param amount                   number of cards to exile
+     * @param toUniqueExileZone        moves the card to a source object dependant
+     *                                 unique exile zone, so another effect of the same source object (e.g.
+     *                                 Theater of Horrors) can identify the card
+     * @param faceDown                 if true, cards are exiled face down
+     * @param withFaceDownReminderText if true, add the reminder text for exiling one face down card
+     */
+    public ExileCardsFromTopOfLibraryControllerEffect(int amount, boolean toUniqueExileZone, boolean faceDown, boolean withFaceDownReminderText) {
+        super(Outcome.Exile);
+        this.amount = amount;
+        this.toUniqueExileZone = toUniqueExileZone;
+        this.faceDown = faceDown;
+        this.withFaceDownReminderText = withFaceDownReminderText;
```

Comment:
I think you no longer need to store this as a field, since it's only used to set staticText?

Reply:
Whoops. You're right, this is unnecessary now that the staticText is set in the constructor.